xauc/probability.py

```diff
@@ -0,0 +1,75 @@
+# Lint as: python3
+"""Functions for computing probabilities used for calculating xAUC metrics.
+
+    Copyright 2020 Google LLC
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        https://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+"""
+
+import numpy as np
+from sklearn import metrics
+
+from labels import validate_labels
+
+def prob_ranked_corectly(scores_a, scores_b, label_a, label_b):
+    """Returns the probability that the scores with be ranked correctly.
+
+    This function uses sklearn's AUC function to calculate the probability that
+    scores_a and scores_b are ranked appropriately relative to each other.
+
+    The scores vector we input to the AUC function is just the concatenation of
+    scores_a and scores_b, while the labels vector we input to the AUC function
+    is label_a for the first len(scores_a) items and label_b for the rest of the
+    vector. The resulting probability is the return value.
+
+    Args:
+        scores_a: a vector of scores with ground truth label label_a
+        scores_b: a vector of scores with ground truth label label_b
+        label_a: the label for scores in scores_a
```

Comment:
Are label_a and label_b two scalars? Do we have any restrictions on them? Like they need to be different, and from {0, 1} and scores of 1 should be higher than scores of 0 for an 'appropriate ranking'? In other words, this would be our regular AUC?

Reply:
I'm doing a refactor per. our discussions of the way the API should act, and in this refactor I'm removing this function, so this shouldn't be a problem.